satellite/accounting/tally/tally.go

```diff
@@ -38,204 +34,185 @@ type Config struct {
 	return &Service{
-		logger:                  logger,
-		metainfo:                metainfo,
-		overlay:                 overlay,
-		Loop:                    *sync2.NewCycle(interval),
+		log:  log,
+		Loop: *sync2.NewCycle(interval),
+
+		metainfoLoop:            metainfoLoop,
+		liveAccounting:          liveAccounting,
 		storagenodeAccountingDB: sdb,
 		projectAccountingDB:     pdb,
-		liveAccounting:          liveAccounting,
 	}
 }
 
 // Run the tally service loop
-func (t *Service) Run(ctx context.Context) (err error) {
+func (service *Service) Run(ctx context.Context) (err error) {
 	defer mon.Task()(&ctx)(&err)
-	t.logger.Info("Tally service starting up")
+	service.log.Info("Tally service starting up")
 
-	return t.Loop.Run(ctx, func(ctx context.Context) error {
-		err := t.Tally(ctx)
+	return service.Loop.Run(ctx, func(ctx context.Context) error {
+		err := service.Tally(ctx)
 		if err != nil {
-			t.logger.Error("tally failed", zap.Error(err))
+			service.log.Error("tally failed", zap.Error(err))
 		}
 		return nil
 	})
 }
 
 // Close stops the service and releases any resources.
-func (t *Service) Close() error {
-	t.Loop.Close()
+func (service *Service) Close() error {
+	service.Loop.Close()
 	return nil
 }
 
 // Tally calculates data-at-rest usage once
-func (t *Service) Tally(ctx context.Context) (err error) {
+func (service *Service) Tally(ctx context.Context) (err error) {
 	defer mon.Task()(&ctx)(&err)
+
 	// The live accounting store will only keep a delta to space used relative
 	// to the latest tally. Since a new tally is beginning, we will zero it out
 	// now. There is a window between this call and the point where the tally DB
 	// transaction starts, during which some changes in space usage may be
 	// double-counted (counted in the tally and also counted as a delta to
 	// the tally). If that happens, it will be fixed at the time of the next
 	// tally run.
-	t.liveAccounting.ResetTotals()
+	service.liveAccounting.ResetTotals()
 
-	var errAtRest, errBucketInfo error
-	latestTally, nodeData, bucketData, err := t.CalculateAtRestData(ctx)
+	// Fetch when the last tally happened so we can roughly calculate the byte-hours.
+	lastTime, err := service.storagenodeAccountingDB.LastTimestamp(ctx, accounting.LastAtRestTally)
 	if err != nil {
-		errAtRest = errs.New("Query for data-at-rest failed : %v", err)
-	} else {
-		if len(nodeData) > 0 {
-			err = t.storagenodeAccountingDB.SaveTallies(ctx, latestTally, nodeData)
-			if err != nil {
-				errAtRest = errs.New("Saving storage node data-at-rest failed : %v", err)
-			}
+		return Error.Wrap(err)
+	}
+	if lastTime.IsZero() {
+		lastTime = time.Now()
+	}
+
+	// add up all nodes and buckets
+	observer := NewObserver()
+	err = service.metainfoLoop.Join(ctx, observer)
+	if err != nil {
+		return Error.Wrap(err)
+	}
+	finishTime := time.Now()
```

Comment:
If i understand that correctly, we will have problems at some point, if the loop takes longer than 24 hours.
We are unable to account the storage for nodes within that timeframe then...

That has not changed with this PR, but made me aware of it.

Reply:
Why would it cause problems?

Reply:
Accounting is getting more vague in terms of we pay them for files that could be deleted already by the time we account it.

Reply:
Yes, that is the case.

Reply:
Thats an issue imo and another reason to keep the table size small and set ourself a maximum iteration time, possibly log the time it took to do a full iteration?